grid/chat/messages/maps/discord.suma

```diff
@@ -0,0 +1,276 @@
+profile = "chat/messages@1.0"
+provider = "discord"
+
+"""
+GetMessages map
+"""
+map GetMessages {
+	destination = input.destination
+	afterId = undefined
+	beforeId = undefined
+
+	set if (input.afterTimestamp) {
+		afterId = call GetSnowflake(ts = input.afterTimestamp)
+	}
+
+	set if (input.beforeTimestamp) {
+		beforeId = call GetSnowflake(ts = input.beforeTimestamp)
+	}
+
+	http GET "/channels/{destination}/messages" {
+		request "application/json" {
+			query {
+				after = afterId
+				before = input.page || beforeId
+				limit = input.limit
+			}
+
+			headers {
+				"Authorization" = `Bot ${parameters.accessToken}`
+			}
+		}
+
+		response 200 "application/json" {
+			messages = call MapMessages(messages = body)
+			rateLimit = call MapRateLimitDetails(headers = headers)
+
+			map result {
+				nextPage = messages[messages.length - 1].id
+				messages = messages
+				rateLimit = rateLimit
+			}
+		}
+
+		response {
+			error = call MapDiscordError(
+				statusCode = statusCode,
+				error = body,
+				headers = headers,
+			)
+			
+			map error error
+		}
+	}
+}
+
+operation MapMessage {
+	index = args.index
+	message = args.messages[index]
+
+	messageResult = {
+		id: message.id,
+		author: {
+			id: message.author.id,
+			username: message.author.username
+		},
+		createdAt: Date.parse(message.timestamp),
+		text: message.content
+	}
+
+	set if (message.edited_timestamp) {
+		messageResult.updatedAt = Date.parse(message.edited_timestamp)
+	}
+
+	set if (message.reactions && message.reactions.length > 0) {
+		messageResult.reactions = message.reactions.map(reaction => ({
+			emoji: reaction.emoji.name,
+			count: reaction.count,
+		}))
+	}
+
+	set if (message.thread) {
+		messageResult.hasThread = true
+		messageResult.threadId = message.thread.id
+	}
+
+	set if (message.attachments && message.attachments.length > 0) {
+		messageResult.attachments = call foreach(attachment of message.attachments) MapAttachment(
+			attachment = attachment
+		)
+	}
+	
+	set if (message.message_reference && message.message_reference.message_id) {
+		messageResult.parentId = message.message_reference.message_id
+	}
+
+	messageResult.url = call ComposeURL(dest = message.channel_id, id = message.id)
+	
+	return messageResult
+}
+
+operation MapAttachment {
+	attachment = args.attachment
+	createdAt = call GetTimestamp(id = attachment.id)
+
+	return {
+		id: attachment.id,
+		createdAt: createdAt,
+		fileName: attachment.filename,
+		mediaType: attachment.content_type,
+		url: attachment.url,
+	}
+}
+
+"""
+Operation MapMessages
+Returns array mapping messages in received body
+"""
+operation MapMessages {
+	listOfMessages = []
+	
+	// workaround for map interpreter variables merging issue
+	listOfMessages = call foreach(index of Array(args.messages.length).fill().map((_, i) => i)) MapMessage(
```

Comment:
@martinalbert This will cause the same performance  problems we had in `social-media/posts-lookup` Twitter map. See my PR https://github.com/superfaceai/station/pull/207/files. The solution is to use single Jessie script for response mapping.

Reply:
Not exactly the workaround, the problem is that OneSDK would have to do thousands of Jessie script evals and the overhead with initialisation of Jessie script eval is high.

Reply:
Thanks! I'll refactor it.

Reply:
I wonder how would sdk perform if there would be a variable instead of raw `Array(args)...` expression. 🤔

Reply:
> I wonder how would sdk perform if there would be a variable instead of raw Array(args)... expression. 🤔

IMO there won't be any difference in the performance.

Reply:
@janhalama will you solution work when I iterate on top of array of returned items in body or it's important to use `[body].map(...`?

Reply:
You do not need to access properties from outside of the array you do not need to use the `[body].map(...` construct.

This should work well:
```
listOfMessages = args.messages.map(message, => {
...
}
```

Reply:
Alright, thanks

Reply:
@janhalama I refactored it, check it out if it's good enough please.

Reply:
looks good